### PR TITLE
Bank raise on unsupported kwargs

### PR DESCRIPTION
**Pull request by d-chambers**
This PR ensures both `EventBank` and `WaveBank` raise a newly defined `UnsupportedKerword` exception when, well, an unsupported keyword is used.

Addresses #207

### CODE REVIEW

**Comment by d-chambers** [comment]
@sboltz mind giving this a quick look-over? The changes in codecov come from deleting code that was run but wasn't actually doing anything.

File: obsplus/utils/bank.py [hunk 1]
```diff
@@ -191,8 +190,33 @@ def __call__(self, starttime, endtime, buffer, **kwargs):
...
+        return starttime, endtime
+
+    def _validate_kwargs(self, kwargs):
+        """Ensure kwargs are supported."""
+        kwarg_set = set(kwargs)
+        if not kwarg_set.issubset(READ_HDF5_KWARGS):
+            bad_kwargs = kwarg_set - set(READ_HDF5_KWARGS)
+            msg = (
+                f"The following kwargs are not supported: {bad_kwargs}. "
+                f"Supported kwargs are {kwarg_set}"
```
Comment (sboltz) [inline obsplus/utils/bank.py:214]: This should be`kwarg_set - bad_kwargs`?

File: tests/test_bank/test_eventbank.py [hunk 2]
```diff
@@ -643,7 +649,7 @@ def test_from_path(self, tmpdir):
 
     def test_put_event_no_reference_time(self, ebank):
         """Test that putting an event with no reference time raises."""
-        # get an event with no reference time and no id
+        # get an event with no reference time and no idm
```
Comment (sboltz) [inline tests/test_bank/test_eventbank.py:652]: What is an idm? Was that a typo?

**Review by sboltz — COMMENTED** [review]
Hi @d-chambers, this looks good to me. The only thing that I noticed is that the error message when you input a bad kwarg is incorrect (the list of supported kwargs will still contain the bad kwarg). I think I might suggest removing that part of the error message entirely, since it won't provide a comprehensive list of acceptable kwargs.

File: tests/test_bank/test_eventbank.py (hunk 2 above)
Comment (d-chambers) [reply tests/test_bank/test_eventbank.py:652]: yes, its a typo.

**Pull request merged**